apps/antalmanac/src/components/Map/Routes.tsx

```diff
@@ -0,0 +1,143 @@
+import { useEffect } from 'react'
+import L from 'leaflet'
+import type { LatLngTuple } from 'leaflet'
+import { useMap } from 'react-leaflet'
+import 'leaflet-routing-machine'
+import { createElementHook, createElementObject, useLeafletContext } from '@react-leaflet/core'
+import type { LeafletContextInterface } from '@react-leaflet/core'
+
+const ACCESS_TOKEN = 'pk.eyJ1IjoicGVkcmljIiwiYSI6ImNsZzE0bjk2ajB0NHEzanExZGFlbGpwazIifQ.l14rgv5vmu5wIMgOUUhUXw';
+
+interface Props {
+  /**
+   * Waypoints needs to be `L.Routing.Waypoint[]` or `LatLng[]` when creating an `L.Routing.plan`.
+   * For ease of use from outside, pass in a valid `LatLngTuple[]`, and convert to LatLng inside.
+   * @example [[33.6405, -117.8443], [33.6405, -117.8443]]
+   */
+  latLngTuples: LatLngTuple[]
+
+  /**
+   * Color of line for this route.
+   */
+  color?: string
+}
+
+/**
+ * Use react-leaflet's core API to manage lifecycle of leaflet elements properly.
+ * @see {@link https://react-leaflet.js.org/docs/core-architecture/#element-hook-factory}
+ */
+function createRouter(props: Props, context: LeafletContextInterface) {
+  const latLngTuples = props.latLngTuples || []
+
+  /**
+   * Convert each tuple to an actual `LatLng` object.
+   */
+  const waypoints = latLngTuples.map((latLngTuple) => L.latLng(latLngTuple))
+
+  const routerControl = L.Routing.control({
+    router: L.Routing.mapbox(ACCESS_TOKEN, {
+      /**
+       * Default is mapbox/driving. More options:
+       * @see {@link https://docs.mapbox.com/api/navigation/directions/#routing-profiles}
+       */
+      profile: 'mapbox/walking',
+    }),
+
+    /**
+     * When searching a location, it yanks control of the map;
+     * not turning this off will screw that up since it refuses to allow the routes to go offscreen.
+     */
+    fitSelectedRoutes: false,
+
+    plan: L.Routing.plan(waypoints, {
+      addWaypoints: false,
+      createMarker: () => false,
+    }),
+
+    routeLine(route) {
```

Comment:
Just calling a function in here makes it confusing to read. Can you take this out, name it, and pass it in here?

Reply:
Ok